File: hw/chavez/u3/hw21AnimalFarm/javascript/controller/AnimalController.js
import Pig from "../model/Pig.js";
import Cow from "../model/Cow.js";
import Chicken from "../model/Chicken.js";
import Sheep from "../model/Sheep.js";
import Food from "../model/Food.js";

export default class AnimalController {

    static run() {

        const food = new Food("Corn");

        console.log("\n----- PIG -----");
        const pig = new Pig(1, "Yorkshire", "2025-01-15");
        pig.feed(food);
        pig.sendToSlaughterHouse("Juan");
        pig.cut();

        console.log("\n----- COW -----");
        const cow = new Cow(2, "Holstein", "2024-05-20");
        cow.feed(food);
        cow.sendToSlaughterHouse("Mario");
        cow.cut();

        console.log("\n----- CHICKEN -----");
        const chicken = new Chicken(3, "Leghorn", "2025-02-10");
        chicken.feed(food);
        chicken.produce();

        console.log("\n----- SHEEP -----");
        const sheep = new Sheep(4, "Mar", "2024-09-18");
        sheep.feed(food);
        sheep.produce();
    }

}

File: hw/chavez/u3/hw21AnimalFarm/python/controller/AnimalController.py
from model.Food import Food
from model.Pig import Pig
from model.Cow import Cow
from model.Chicken import Chicken
from model.Sheep import Sheep


class AnimalController:

    @staticmethod
    def run():

        food = Food("Corn")

        print("\n----- PIG -----")
        pig = Pig(1, "Yorkshire", "2025-01-15")
        pig.feed(food)
        pig.send_to_slaughter_house("Don Pepe")
        pig.cut()

        print("\n----- COW -----")
        cow = Cow(2, "Holstein", "2024-05-20")
        cow.feed(food)
        cow.send_to_slaughter_house("Don Pepe")
        cow.cut()

        print("\n----- CHICKEN -----")
        chicken = Chicken(3, "Leghorn", "2025-02-10")
        chicken.feed(food)
        chicken.produce()

        print("\n----- SHEEP -----")
        sheep = Sheep(4, "Merino", "2024-09-18")
        sheep.feed(food)
        sheep.produce()